src/support/utf8.c

```diff
@@ -615,71 +615,63 @@ size_t u8_printf(const char *fmt, ...)
     return cnt;
 }
 
-/* based on the valid_utf8 routine from the PCRE library by Philip Hazel
+/* Rewritten completely, original code not based on anything else
 
    length is in bytes, since without knowing whether the string is valid
    it's hard to know how many characters there are! */
-int u8_isvalid(const char *str, size_t length)
+int u8_isvalid(const char *iStr, size_t iLength)
```

Comment:
Yes you are right about variable naming. I just wanted to reply to your complaint about formatting.

I can absolutely see some of your other points. For instance it would be a good move to put globals into structs so that there is a julia context that can be independently hold. But one has also to acknowledge that the C core of julia is moving fast and maintained/developed by a small group of people. Taking this into account development has been pretty efficient IMHO.

If you want to help improving the C core I suggest opening a new issue and making concrete proposals that the core hackers can judge.